### PR TITLE
Update the documentation for port stats retrieval using gnmi-cli.

### PR DESCRIPTION
**Pull request by np2-git**
Title: Update the documentation for port stats retrieval using gnmi-cli.
Signed-off-by: np2-git <narendra.gupta.p@intel.com>

### CODE REVIEW

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-62-1/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/0a21cf066725dcc7db6d0ab7ebb7dcfcd7186200

File: Documentation/howto/ovs-with-p4-executables.rst [hunk 1]
```diff
@@ -661,6 +661,33 @@ previously configured CONFIG params.
...
+    $ Example for LINK ports :
+    gnmi-cli get "device:physical-device,name:PORT0,counters" | grep "name\|uint_val" | grep -v "interface\|key\|config\|counters"
+    $ Example for vhost ports :
+    gnmi-cli get "device:virtual-device,name:net_vhost0,counters" | grep "name\|uint_val" | grep -v "interface\|key\|config\|counters"
+    $ Example for TAP ports :
+    gnmi-cli get "device:virtual-device,name:TAP0,counters" | grep "name\|uint_val" | grep -v "interface\|key\|config\|counters"
+
+  .. note::
+
+    Port stats can be retrieved for the ports that are configured through GNMI CLI.
```
Comment (nupuruttarwar) [inline Documentation/howto/ovs-with-p4-executables.rst]: created through GNMI CLI? And can we also add " PORT0, net_vhost0 and TAP0 corresponds to the name used when creating the ports using GNMI CLI"?

File: Documentation/howto/ovs-with-p4-executables.rst [hunk 2]
```diff
@@ -661,6 +661,33 @@ previously configured CONFIG params.
...
+    These ports can be of type LINK/VHOST/TAP type. gnmi-cli by default outputs
+    the data in yang model, hence format the output using the grep comamnd for
```
Comment (nupuruttarwar) [inline Documentation/howto/ovs-with-p4-executables.rst]: so the output is formatted using the grep command to display port statistics

File: Documentation/howto/ovs-with-p4-executables.rst [hunk 3]
```diff
@@ -661,6 +661,33 @@ previously configured CONFIG params.
     tdi-portin-id: Port ID for Pipeline in Input Direction
     tdi-portin-id: Port ID for Pipeline in Output Direction
 
+10) Get atrributes for port statistics retrieval::
```
Comment (nupuruttarwar) [inline Documentation/howto/ovs-with-p4-executables.rst]: Get port statistics for VHOST, Physical link and non-control TAP ports?

File: Documentation/howto/ovs-with-p4-executables.rst [hunk 4]
```diff
@@ -661,6 +661,33 @@ previously configured CONFIG params.
...
+
+  .. note::
+
+    Port stats can be retrieved for the ports that are configured through GNMI CLI.
+    These ports can be of type LINK/VHOST/TAP type. gnmi-cli by default outputs
+    the data in yang model, hence format the output using the grep comamnd for
+    pin-pointed dsplay.
+
+    ``gnmi-cli get`` counters command doesn't work for the TAP ports that are
+    added as control ports. But these control ports when added to OVS bridge
```
Comment (nupuruttarwar) [inline Documentation/howto/ovs-with-p4-executables.rst]: is not applicable for TAP ports that are  added as control ports. Can we create a separate section for Port statistic retrieval for control TAP ports and we can just say that refer to the section ".." to retrieve port statistics for control TAP ports? Thoughts?

File: Documentation/howto/ovs-with-p4-executables.rst [hunk 5]
```diff
@@ -661,6 +661,33 @@ previously configured CONFIG params.
...
+    through the "ovs-vsctl add-port <BRIDGE> <TAP-PORT>' command, stats can be
+    read through the 'ovs-ofctl dump-ports <BRIDGE>" command
+
+    ``PARAMS``: These params are key:value pairs. Here physical-device or
+    virtual-device is a sub-node which holds multiple ports,... Pass
```
Comment (nupuruttarwar) [inline Documentation/howto/ovs-with-p4-executables.rst]: ,.... to .

**Review by nupuruttarwar — CHANGES_REQUESTED** [review]

**Comment by nupuruttarwar** [comment]
Please check why build is failing

File: Documentation/howto/ovs-with-p4-executables.rst (hunk 3 above)
Comment (np2-git) [reply Documentation/howto/ovs-with-p4-executables.rst]: Done. Taken care.

File: Documentation/howto/ovs-with-p4-executables.rst (hunk 5 above)
Comment (np2-git) [reply Documentation/howto/ovs-with-p4-executables.rst]: done

File: Documentation/howto/ovs-with-p4-executables.rst (hunk 4 above)
Comment (np2-git) [reply Documentation/howto/ovs-with-p4-executables.rst]: corrected - **is not applicable for TAP ports**.  Also created separate section for 'control TAP ports'

File: Documentation/howto/ovs-with-p4-executables.rst (hunk 3 above)
Comment (np2-git) [reply Documentation/howto/ovs-with-p4-executables.rst]: Done

File: Documentation/howto/ovs-with-p4-executables.rst (hunk 2 above)
Comment (np2-git) [reply Documentation/howto/ovs-with-p4-executables.rst]: Done

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-62-2/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/093c623f293e8a93d4f190f5424141699f308af1

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-62-3/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/093c623f293e8a93d4f190f5424141699f308af1

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-62-4/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/15554af83be40fdfa6b0e008ce4b940776c1343a

File: Documentation/howto/ovs-with-p4-executables.rst (hunk 1 above)
Comment (np2-git) [reply Documentation/howto/ovs-with-p4-executables.rst]: Done

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-62-5/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/4af0ee5b92171a15b541f6453b75d104d4008619

**Comment by np2-git** [comment]
> Please check why build is failing

This is passing now

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-62-6/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/4af0ee5b92171a15b541f6453b75d104d4008619

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-62-7/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/4af0ee5b92171a15b541f6453b75d104d4008619

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-62-8/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/4af0ee5b92171a15b541f6453b75d104d4008619

**Review by nupuruttarwar — APPROVED** [review]
LGTM

**Review by 5abeel — APPROVED** [review]
lgtm

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-62-9/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/7a9df1db3174f94fa98ce1759786e6924cc8a62a

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-62-10/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/7a9df1db3174f94fa98ce1759786e6924cc8a62a

**Comment by ipdksw** [comment]
Result:FAILURE JenkinsConsole:https://s3.ipdk.io/ovs/PR-62-11/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/7a9df1db3174f94fa98ce1759786e6924cc8a62a

**Pull request merged**